ir/src/main/scala/org/scalajs/core/ir/Serializers.scala

```diff
@@ -720,14 +729,27 @@ object Serializers {
           // read and discard the length
           val len = readInt()
           assert(len >= 0)
-          val result = MethodDef(readBoolean(), readPropertyName(),
+          val result1 = MethodDef(readBoolean(), readPropertyName(),
               readParamDefs(), readType(), readTree())(
               new OptimizerHints(readInt()), optHash)
-          if (foundArguments) {
+          val result2 = if (foundArguments) {
             foundArguments = false
-            new RewriteArgumentsTransformer().transformMethodDef(result)
+            new RewriteArgumentsTransformer().transformMethodDef(result1)
           } else {
-            result
+            result1
+          }
+          if (useHacks065 && result2.resultType != NoType &&
+              isConstructorName(result2.name.name)) {
+            val patchedBody = result2.body match {
+              case This()                 => Skip()(result2.body.pos)
+              case Block(stats :+ This()) => Block(stats)(result2.body.pos)
+              case body                   => body
+            }
```

Comment:
Ok. Maybe drop a comment. It is hard to understand why you need to do this (in a hack), but using the escape hatch (just not patching the body) is fine as well.